cmd/clair/main.go

```diff
@@ -201,12 +203,36 @@ func Boot(config *Config, slimMode bool) {
 	serv.Close()
 }
 
+func bootNodeInventoryScanner() {
+	// TODO: Check: Unify with Scanner boot and run as a different configuration rather than a different function.
+	grpcAPI := grpc.NewAPI(
+		grpc.WithCustomRoutes(debugRoutes),
+		grpc.WithCustomUnaryInterceptors(grpcprometheus.UnaryServerInterceptor))
+
+	grpcAPI.Register(
+		ping.NewService(),
+		nodeinventory.NewService(env.NodeName.Value()),
+	)
+	go grpcAPI.Start()
+
+	// Wait for interruption and shutdown gracefully.
+	waitForSignals(os.Interrupt, unix.SIGTERM)
+	log.Info("Received interruption, gracefully stopping ...")
+}
+
 func main() {
 	// Parse command-line arguments
 	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
 	flagConfigPath := flag.String("config", "/etc/scanner/config.yaml", "Load configuration from the specified file.")
+	flagNodeInventoryMode := flag.Bool("nodeinventory", false, "Run Scanner binary in Node Inventory mode (this should only be used in Secured Clusters)")
```

Comment:
As we are ignoring the config while `nodeinventory` is set to true: maybe we should show a warning when the user tries to provide both flags nodeinventory and config?

Reply:
On skipping config: I see that we will use unconfigured logger when we skip reading the config file here. This is not a big issue as the behavior will be consistent within the nodeInventory mode, but maybe we want to do the configuration, as scanner uses structured logs?

Reply:
Yea, this is my point (2) for reviewers on the PR description. I think it does make sense that some startup routines are shared between the original scanner mode and node inventory. But I didn't want to refactor too much without Scanner's team feedback on this. 

I agree it might make sense to _at least_ start the logger for both. However, the log level is defined in the `config.yaml`. Which means that in node inventory mode we'd also rely on the config file, even if most of the properties aren't used.

Reply:
Out of curiosity: 
Could we initialize the logger with defaults and overwrite them if there is a config found (later on)?
That way, we wouldn't be depending on scanners `config.yaml` but still act on it, e.g. when the log level is set to debug by someone.

Reply:
The logger has its defaults I think, so there is no danger here. Maybe we can configure the logger additionally (when `flagNodeInventoryMode` is true) with the default values from the config?

I do not like that starting the nodeInventoryMode would require a config file... I think we should avoid that.

Reply:
That makes sense. Moved the inventory mode startup. Thanks!